sdk/typescript/test/e2e/encrypt-user-share.test.ts

```diff
@@ -0,0 +1,176 @@
+			toolbox.client,
+			toolbox.keypair,
+			encryptedUserShareAndProof,
+			publicKeyID,
+			dwalletID,
+		);
+
+		const decryptedKeyShare = decrypt_user_share(
+			encryptionKey,
+			decryptionKey,
+			encryptedUserShareAndProof,
+		);
+
+		let secretUserShare = new Uint8Array(256);
+		secretUserShare.set(secretShare.reverse());
+		expect(decryptedKeyShare).toEqual(secretUserShare);
+
+		expect(
+			verify_user_share(
+				// Take the first 32 bytes, the only ones that are non-zero, and reverse them to convert them
+				// from little-endian encoding to big-endian.
+				// This is because of BCS and PlaintextSpaceGroupElement serialization.
+				// PlaintextSpaceGroupElement is U2048 and has 32LIMBS of 64 bits each.
+				new Uint8Array(decryptedKeyShare.slice(0, 32).reverse()),
+				new Uint8Array(dwallet?.decentralizedDKGOutput!),
+			),
+		).toBeTruthy();
+	});
+
+	it('creates an encryption key & stores it in the active encryption keys table', async () => {
+		const [encryptionKey, _] = generate_keypair();
+		const pubKeyRef = await storeEncryptionKey(
+			encryptionKey,
+			EncryptionKeyScheme.Paillier,
+			toolbox.keypair,
+			toolbox.client,
+		);
+		console.log({ pubKeyRef });
+
+		const encryptionKeysHolder = await createActiveEncryptionKeysTable(
+			toolbox.client,
+			toolbox.keypair,
+		);
+
+		await setActiveEncryptionKey(
+			toolbox.client,
+			toolbox.keypair,
+			pubKeyRef?.objectId!,
+			encryptionKeysHolder.objectId,
+		);
+
+		const activeEncryptionKeyAddress = await getActiveEncryptionKeyObjID(
+			toolbox.client,
+			toolbox.keypair.toSuiAddress(),
+			encryptionKeysHolder.objectId,
+		);
+
+		expect(`0x${activeEncryptionKeyAddress}`).toEqual(pubKeyRef?.objectId!);
+	});
+
+	it('encrypts a secret share to a given Sui address successfully', async () => {
+		const [encryptionKey, decryptionKey] = generate_keypair();
+		const pubKeyRef = await storeEncryptionKey(
+			encryptionKey,
+			EncryptionKeyScheme.Paillier,
+			toolbox.keypair,
+			toolbox.client,
+		);
+		console.log({ pubKeyRef });
+
+		const dkg = await createDWallet(toolbox.keypair, toolbox.client);
+
+		const encryptionKeysHolder = await createActiveEncryptionKeysTable(
+			toolbox.client,
+			toolbox.keypair,
+		);
+
+		await setActiveEncryptionKey(
+			toolbox.client,
+			toolbox.keypair,
+			pubKeyRef?.objectId!,
+			encryptionKeysHolder.objectId,
+		);
+
+		const publicKeyID = pubKeyRef?.objectId;
+		const recipientData = await getEncryptionKeyByObjectId(toolbox.client, publicKeyID);
+		const secretShare = dkg?.secretKeyShare!;
+		const encryptedUserShareAndProof = generate_proof(
+			new Uint8Array(secretShare),
+			recipientData?.encryptionKey!,
+		);
+
+		await sendUserShareToAddress(
+			toolbox.client,
+			toolbox.keypair,
+			dkg!,
+			toolbox.keypair.getPublicKey(),
+			encryptionKeysHolder.objectId,
+		);
+
+		const decryptedKeyShare = decrypt_user_share(
+			encryptionKey,
+			decryptionKey,
+			encryptedUserShareAndProof,
+		);
```

Comment:
You are using the wrong `encryptedUserShareAndProof`; you should use the one that you are storing on chain from `sendUserShareToAddress`, not a new one that you are generating in this function as you do